include/pybind11/pytypes.h

```diff
@@ -1640,6 +1640,9 @@ class str : public object {
     str(std::u8string_view s) : str(reinterpret_cast<const char *>(s.data()), s.size()) {}
 #    endif
 
+    // Avoid ambiguity when converting from kwargs (GCC)
+    explicit str(const kwargs &k) : str(static_cast<handle>(k)) {}
```

Comment:
Yuck, it's weird because the handle goes through the raw_str function and steals THAT REFERENCE.